lib/explorer/query.ex

```diff
@@ -257,17 +257,120 @@ defmodule Explorer.Query do
 
   ## Implementation details
 
-  Queries simply become lazy dataframe operations at runtime.
-  For example, the following query
+  This section describes what's happening "under the hood" when working with the
+  `Explorer.Query` interface. You don't need to read this section to work with
+  the API as intended. But it may be helpful if you're trouble shooting or are
+  just curious.
 
-      Explorer.DataFrame.filter(df, nums > 2)
+  Macros like `Explorer.DataFrame.mutate` each have a `*_with` function
+  counterpart. When you write `Explorer.DataFrame.mutate(df, expression)`, the
+  compiler rewrites it as `Explorer.DataFrame.mutate_with(df,
+  callback_from_expression)`. This section describes the translation from
+  `expression` to `callback_from_expression` in a bit more detail.
 
-  is equivalent to
+  Let's start with an example.
 
-      Explorer.DataFrame.filter_with(df, fn df -> Explorer.Series.greater(df["nums"], 2) end)
+      df = Explorer.DataFrame.new(a: [1, 2, 3])
 
-  This means that, whenever you want to generate queries programatically,
-  you can fallback to the regular `_with` APIs.
+      # Macro
+      Explorer.DataFrame.mutate(df, b: a + 1)
+
+      # Callback equivalent
+      Explorer.DataFrame.mutate_with(df, fn query_frame ->
+        lazy_column_a = query_frame["a"]
+        lazy_column_b = Explorer.Series.add(lazy_column_a, 1)
+        [b: lazy_column_b]
+      end)
+
+  Although each verb expects a slightly different return value from their
+  callback, all expect a `%Explorer.Backend.QueryFrame{}` as the input. The
```

Comment:
They don't expect a Explorer.Backend.QueryFrame, it is a regular DataFrame, but backed by that given backend.

Reply:
Oops you're right! I forgot it was a query frame "backed" dataframe:

```elixir
iex(2)> df |> Explorer.DataFrame.mutate_with(fn x ->
...(2)> x |> IO.inspect()
...(2)> [b: x["a"] |> Explorer.Series.add(1)]
...(2)> end)
#Explorer.DataFrame<
  QueryFrame[??? x 1]
  a s64
>
#Explorer.DataFrame<
  Polars[3 x 2]
  a s64 [1, 2, 3]
  b s64 [2, 3, 4]
>
```

Will fix.